quill-core/src/main/scala/io/getquill/dsl/MetaDslMacro.scala

```diff
@@ -194,8 +194,10 @@ class MetaDslMacro(val c: MacroContext) {
             tpe match {
               case tpe if !is[Embedded](tpe) && nested =>
                 c.fail(
-                  s"Can't expand nested value '$tpe', please make it an `Embedded` " +
-                    s"case class or provide an implicit $encoding for it."
+                  s"""Can't find implicit `$encoding[$tpe]`. Please:
+                     |1. ensure that implicit `$encoding[$tpe]` is provided and there are no other conflicting implicits;
+                     |2. or make `$tpe` `Embedded` case class or `AnyVal`.
```

Comment:
I wanted to emphasize that these things are mutually exclusive. I can replace "Please:" with "Please, do one of the following things:" and remove "or".